bsip-0032.md

```diff
@@ -0,0 +1,92 @@
+    BSIP: 0032
+    Title: Always Match Orders At Maker Price
+    Author: Abit More <https://github.com/abitmore>
+    Status: Draft
+    Type: Protocol
+    Created: 2018-02-16
+    Discussion: https://github.com/bitshares/bitshares-core/issues/338
+    Replaces: -
+    Worker: To be done
+
+# Abstract
+
+Currently, under most circumstances, when matching two orders, the maker price
+will be used. That said, the order that is placed earlier sets a price,
+another order that is placed later accepts the price, thus a match, two orders
+pay to each other at that price. For example, if one person (A) placed a limit
+order to sell 100 BTS at 0.1 USD per BTS, another person (B) then placed a new
+limit order to buy 100 BTS at 0.105 USD per BTS, the two orders will match at
+0.1 USD per BTS, so A will pay 100 BTS and get 10 USD, B will pay 10 USD and
+get 100 BTS.
+
+However, when matching a taker limit order with a maker margin call order,
+the taker price is being used. For example, if trader A's margin call order is
+selling 100 BTS at no less than 0.1 USD per BTS, then trader B placed an order
+that buys 100 BTS at 0.105 USD per BTS, the two order will match at 0.105 USD
+per BTS, so A will pay 100 BTS and get 10.5 USD, B will pay 10.5 USD and get
+100 BTS.
+
+This BSIP proposes a new behavior: always match orders at maker price.
+
+# Motivation
+
+Make the exchange system more user-friendly.
```

Comment:
Currently, the market engine shows unexpected behaviour in certain situations.
In particular, when placing a limit (e.g. sell) order with a price beyond the highest buy, the order is expected to "walk the book", matching each order on the opposite side in turn, at that order's price, until the new limit order is completely filled, or there are no more buy orders matching its price.
Usually, this expectation is fulfilled. However, when there is a margin call, the margin call will be preferred in matching the incoming order, filling it at its given price. While not strictly a bug, this behaviour is unexpected and irritating for users.

This BSIP proposes more simple order matching rules-

Reply:
This is "abstract" or "motivation"? Looks like abstract, but it's long.. what should we put in the other section?

Reply:
Actually the call-order-take-precedence rule is discussed in another BSIP, so I don't want to mention it here, but only the match price issue.